Update Pagination.md (#859)
wix-incubator/wix-rest-docs: guides/Pagination.md

Most APIs default to sorting by `createdDate` in `DESC` (descending) order,
although some APIs have a different default sort order.
You can often override the default sorting by specifying
a new field and order.

### Sort _List_ endpoints

_List_ endpoints are designed to be lightweight requests.
For this reason, sorting is applied through the query parameters,
typically with sort field name and order fields.

For example, to list contacts by last name in ascending order:

::::tabs
:::REST_TAB
```txt
?sort.fieldName=info.name.last&sort.order=ASC
```
:::
:::SDK_TAB
```
const sort = [
  { fieldName: "info.name.last", order: "ASC" },
]
```
:::
::::

### Sort _Query_ endpoints

_Query_ endpoints offer more robust filtering capabilities.
When working with a _Query_ endpoint,
sorting is specified in REST in an array in the request body,
typically `query.sort`, and in the SDK with `.ascending()` and `.descending()` functions in the SDK's query chain. 

> For each `sort` object,

Author: ```suggestion
For each `query.sort` object,
```

> sorting is applied with the `fieldName` and `order` parameters.

Author: ```suggestion
sorting is typically applied with the `fieldName` and `order` parameters.
```

### Sort _List_ endpoints

_List_ endpoints are designed to be lightweight requests.
For this reason, sorting is applied through the query parameters,
typically with sort field name and order fields.

For example, to list contacts by last name in ascending order:

::::tabs
:::REST_TAB
```txt
?sort.fieldName=info.name.last&sort.order=ASC
```
:::
:::SDK_TAB
```

> const sort = [

Author: I can't find an actual code sample in the SDK that has a sort example :(

  { fieldName: "info.name.last", order: "ASC" },
];
```
:::
::::

Your API's _List_ method may support sorting by multiple fields.
This is done by adding new field name and order parameters
for each sort field:

::::tabs
:::REST_TAB
```txt
?sort.fieldName=info.name.last&sort.order=ASC&sort.fieldName=createdDate&sort.order=ASC
```
:::
:::SDK_TAB
```
const sort = [
  { fieldName: "status", order: "ASC" },
  { fieldName: "createdDate", order: "DESC" }
];

```
:::
::::

### Sort _Query_ endpoints

_Query_ endpoints offer more robust filtering capabilities.
When working with a _Query_ endpoint,
sorting is specified in REST in an array in the request body,
typically `query.sort`, and in the SDK within a .sort() function in the SDK's query chain. 
For each `sort` object,
sorting is applied with the `fieldName` and `order` parameters.

> For example, to list contacts by last name in ascending order,

Author: ```suggestion
For example, to list payment links by created date in ascending order, and by status in descending order:
```

> include this structure in the request body:

Author: ```suggestion
```

### Sort _List_ endpoints

_List_ endpoints are designed to be lightweight requests.
For this reason, sorting is applied through the query parameters,
typically with sort field name and order fields.

For example, to list contacts by last name in ascending order:

::::tabs
:::REST_TAB
```txt
?sort.fieldName=info.name.last&sort.order=ASC
```
:::

> :::SDK_TAB

Author: full code sample:
import { listContacts } from 'wix-crm-backend';

const sort = [
  { fieldName: "info.name.last", order: "ASC" },
];

listContacts({ sort })
  .then(results => {
    // results.items is sorted as requested
    console.log(results.items);
  })
  .catch(error => {
    console.error(error);
  });

### Sort _List_ endpoints

_List_ endpoints are designed to be lightweight requests.
For this reason, sorting is applied through the query parameters,
typically with sort field name and order fields.

For example, to list contacts by last name in ascending order:

::::tabs
:::REST_TAB
```txt
?sort.fieldName=info.name.last&sort.order=ASC
```
:::
:::SDK_TAB
```
import { listContacts } from 'wix-crm-backend';

const sort = [
  { fieldName: "info.name.last", order: "ASC" },
];

> listContacts({ sort })

Author: cut?

  }
}
```
:::
:::SDK_TAB
```
import { paymentLinks } from "@wix/get-paid";

export const queryPaymentLinksSorted = webMethod(
  async () => {
    try {
      const results = await paymentLinks.queryPaymentLinks()
        .ascending('_createdDate')        // Sort by created date ascending
        .descending('status')             // Then by status descending
        .find();

      console.log(`Found ${results.items.length} payment links`);
      console.log('Payment links:', results.items);
      
      return {
        success: true,
        paymentLinks: results.items,
        totalCount: results.totalCount,
        pageSize: results.pageSize,
        sortedBy: 'created date (oldest first), then status (descending)'
      };

      }
    }
);

> ```

Author: import { paymentLinks } from "@wix/get-paid";
export const queryPaymentLinksSorted = webMethod(
  async () => {
    try {
      const results = await paymentLinks.queryPaymentLinks()
        .ascending('_createdDate')        // Sort by created date ascending
        .descending('status')             // Then by status descending
        .find();
      console.log(`Found ${results.items.length} payment links`);
      console.log('Payment links:', results.items);
      
      return {
        success: true,
        paymentLinks: results.items,
        totalCount: results.totalCount,
        pageSize: results.pageSize,
        sortedBy: 'created date (oldest first), then status (descending)'
      };
      }
    }
);

- **Offset-based** (includes `limit` and `skip`/`offset`): Specify the number of items to skip and the number to return.
- **Cursor-based** (includes `nextCursor` and `prevCursor`): Use a token (cursor) to fetch the next or previous set of results.

### Paging _List_ endpoints

For example, to list 100 contacts, starting from contact 20, with offset paging:

::::tabs
:::REST_TAB
```txt
?limit=100&offset=20
```
:::
:::SDK_TAB
```
{
    "options": {
       "paging": {
           "limit": 20,
           "offset": 0
       },
    }
 }
```
:::
::::

To list the next 100 contacts with cursor paging:

::::tabs
:::REST_TAB
```txt
&cursorPaging.cursor=JWE.eyJhbGciOiJBMTI4S1ciLCJlbm
```
:::
:::SDK_TAB
```
{
    const options = {
       cursorPaging {
            cursor: JWE.eyJhbGciOiJBMTI4S1ciLCJlbm,
    }
}
};
```
:::
::::

### Paging _Query_ endpoints

For example, to query 100 contacts, starting from contact 20, with offset paging:

::::tabs
:::REST_TAB
```json
  "query": {
    "paging": {
    "limit": 100, 
    "offset": 20
  }
}
```
:::
:::SDK_TAB
```

> import { contacts } from "@wix/crm";

Author: export const queryContactsSorted = webMethod(
  Permissions.Anyone,  // or Permissions.SiteMember
  async () => {
    try {
      const results = await contacts.queryContacts()
        .limit(10)  
        .skip(20)
        .ascending('info.name.last')        // Sort by last name (A–Z)
        .find();

      console.log(`Found ${results.items.length} contacts`);
      console.log('Contacts:', results.items);
      
      return {
        success: true,
        contacts: results.items,
        totalCount: results.totalCount,
        pageSize: results.pageSize
      };
      
    } catch (error) {
      console.error('Error querying contacts:', error);
      return {
        success: false,
        error: error.message
      };
    }
  }
);

const query = contacts.queryContacts()               
  .limit(100)  
  .skip(20)
  .find()
  .then(results => {
    console.log(results.items); // Sorted and filtered results
  })
  .catch(error => {
    console.error(error);
  });
```
:::
::::

Should return items 21-120 in the results.

To query 100 payment links with cursor paging:

::::tabs
:::REST_TAB
```json
  "query": {
    "cursorPaging": {
      "cursor": JWE.eyJhbGciOiJBMTI4S1ciLCJlbm,
    }
```
:::

> :::SDK_TAB

Author: I haven't managed to get this one to work on my test site

  })
  .catch(error => {
    console.error(error);
  });
:::
::::

## Paging

Paging allows you to control how many results are returned and where the result set starts. Wix APIs commonly support 1 or both of 2 paging strategies:

- **Offset-based** (limit/skip/offset): Specify the number of items to skip and the number to return.
- **Cursor-based** (nextCursor): Use a token (cursor) to fetch the next set of results.

### Paging _List_ endpoints

For example, to list 100 contacts, starting from contact 20, with offset paging:

::::tabs
:::REST_TAB
```txt
?limit=100&offset=20
```
:::
:::SDK_TAB
```
{
    "options": {
       "paging": {
           "limit": 20,
           "offset": 0
       },
    }
 }
```
:::
::::

To list the next 100 contacts with cursor paging:

::::tabs
:::REST_TAB
```txt
&cursorPaging.cursor=JWE.eyJhbGciOiJBMTI4S1ciLCJlbm
```
:::
:::SDK_TAB
```
{
    const options = {
       cursorPaging {
            cursor: JWE.eyJhbGciOiJBMTI4S1ciLCJlbm,
    }
}
};
```
:::
::::

### Paging _Query_ endpoints

For example, to query 100 contacts, starting from contact 20:

::::tabs
:::REST_TAB
```json
    "limit": 100, 
    "offset": 20 
```
:::
:::SDK_TAB

> import { contacts } from 'wix-crm-backend'; 

Author: ```suggestion
import { contacts } from "@wix/crm";
```

  }
}
```
:::
:::SDK_TAB
import { contacts } from 'wix-crm-backend';

contacts.query()       
  .ascending('info.name.last')                 // Sort results by last name (A–Z)
  .descending('createdDate')                   // Then by creation date (newest first)
  .find()
  .then(results => {
    console.log(results.items); // Sorted and filtered results
  })
  .catch(error => {
    console.error(error);
  });
:::
::::

## Paging

The standard Wix API pagination includes:

- **limit**: amount of items per response (default is `0`)

- **offset/skip**: number of items to skip. The standard in REST is **offset**, and in the SDK is **skip**.

- **cursor paging**: 

### Paging _List_ endpoints

For example, to list 100 contacts, starting from contact 20:

::::tabs
:::REST_TAB
```txt
?limit=100&offset=20
```
:::

> :::SDK_TAB

Author: full code sample:
import { listContacts } from 'wix-crm-backend';

const sort = [
  { fieldName: "info.name.last", order: "ASC" },   
  { fieldName: "createdDate", order: "DESC" }      
];

listContacts({
  sort,     
  limit: 100, // Number of items to return
  skip: 20  // Number of items to skip (i.e., start with contact #101)
})
.then(results => {
  console.log(results.items);
})
.catch(error => {
  console.error(error);
});

Author: SDK list method with offset paging: https://dev.wix.com/docs/sdk/backend-modules/members/members/list-members

  .descending('createdDate')                   // Then by creation date (newest first)
  .find()
  .then(results => {
    console.log(results.items); // Sorted and filtered results
  })
  .catch(error => {
    console.error(error);
  });
:::
::::

## Paging

Paging allows you to control how many results are returned and where the result set starts. Wix APIs commonly support 1 or both of 2 paging strategies:

- **Offset-based** (limit/skip/offset): Specify the number of items to skip and the number to return.
- **Cursor-based** (nextCursor): Use a token (cursor) to fetch the next set of results.

### Paging _List_ endpoints

For example, to list 100 contacts, starting from contact 20, with offset paging:

::::tabs
:::REST_TAB
```txt
?limit=100&offset=20
```
:::
:::SDK_TAB
```
{
 *   paging: {
 *       limit: 100,
          skip: 20
 *   }     
];
```
:::
::::

To list the next 100 contacts with cursor paging:

::::tabs
:::REST_TAB
```txt
&cursorPaging.cursor=JWE.eyJhbGciOiJBMTI4S1ciLCJlbm
```
:::
:::SDK_TAB
```
{
    const options = {
       cursorPaging {
            cursor: JWE.eyJhbGciOiJBMTI4S1ciLCJlbm

>     }

Author: ```suggestion
```

Author: SDK list calls with cursor paging:
https://dev.wix.com/docs/sdk/backend-modules/bookings/external-calendars/list-events
https://dev.wix.com/docs/sdk/backend-modules/media/folders/list-folders
